### ISSUE TITLE
 Error:1 of 2 post-start scripts failed. Failed Jobs: uaa. Successful Jobs: bosh-dns.

### ISSUE THREAD

**Issue by forexamplelyr** [cloudfoundry/uaa#931]
Action Failed get_task: Task 26b818c2-fb11-4967-7681-09b534d045cf result: 1 of 2 post-start scripts failed. Failed Jobs: uaa. Successful Jobs: bosh-dns.
uaa/8acb45cd-039c-40d1-8d42-9ea04c57bf68:~# Curl -I 127.0.0.1 :8080
HTTP/1.1  404
The jobs on uaa vms display running;The uaa vms display running; 
************************************************************************************************
I have created env with below commands.
bosh create-env bosh.yml 
--state ./state.json 
-o virtualbox/cpi.yml 
-o virtualbox/outbound-network.yml 
-o bosh-lite.yml 
-o bosh-lite-runc.yml 
-o uaa.yml 
-o credhub.yml 
-o jumpbox-user.yml 
--vars-store ./creds.yml 
-v director_name="bosh-lite" 
-v internal_ip=192.168.50.6 
-v internal_gw=192.168.50.1 
-v internal_cidr=192.168.50.0/24
-v outbound_network_name=NatNetwork

After this, I deployed cf-deployment. Now all VMs' status are running. got below error:
Task 106 | 06:16:54 | Updating instance nats: nats/fd2949bd-9a5c-4bb0-8405-12d95aec70cb (0) (canary) (00:01:55)
Task 106 | 06:16:54 | Updating instance database: database/a93cc713-24a9-4086-bedf-c5a9840e0d77 (0) (canary) (00:08:36)
Task 106 | 06:25:30 | Updating instance diego-api: diego-api/1c1b5cb9-fb30-4885-bd28-28080c05379d (0) (canary) (00:03:00)
Task 106 | 06:28:31 | Updating instance uaa: uaa/8acb45cd-039c-40d1-8d42-9ea04c57bf68 (0) (canary) (00:59:53)
L Error: Action Failed get_task: Task 9cc74522-6b1a-434e-6f31-a5b0505b3c14 result: 1 of 2 post-start scripts failed. Failed Jobs: uaa. Successful Jobs: bosh-dns.
Task 106 | 07:28:25 | Error: Action Failed get_task: Task 9cc74522-6b1a-434e-6f31-a5b0505b3c14 result: 1 of 2 post-start scripts failed. Failed Jobs: uaa. Successful Jobs: bosh-dns.

Here is VMs status:
Instance Process State AZ IPs VM CID VM Type Active
adapter/3bfd9808-d43c-4b10-a61b-d973c91a2ad3 running z1 10.244.0.129 6b584c98-3398-4b6a-6f20-18dfc3f0a915 minimal true
api/d92bdbdd-966f-4af2-bfb5-42071924c445 running z1 10.244.0.134 a31c3888-c8f4-4940-600e-4c3838f81bc7 - true
cc-worker/addd2080-9500-495b-8096-211f62d21598 running z1 10.244.0.135 bcdda010-a447-4682-68ac-c6e4f5563081 - true
credhub/d10f98b9-dd73-4cf2-bd82-3c50d1f12879 running z1 10.244.0.141 8b3383ec-9e3f-4c6a-578a-2de210bf896c minimal true
database/a93cc713-24a9-4086-bedf-c5a9840e0d77 running z1 10.244.0.130 01337e3b-af74-43fc-7f7a-5facaa23db0d small true
diego-api/1c1b5cb9-fb30-4885-bd28-28080c05379d running z1 10.244.0.131 ea8f65e0-7d5b-4534-79a9-464887c7d8d4 small true
diego-cell/2a1bfa5d-8b7b-4b2c-a2b5-2b0fb26cdc27 running z1 10.244.0.139 19332606-a44c-4a4c-67c6-9f8f3939d798 - true
doppler/ea0a2fa7-8e52-4282-a3df-a025d85f214b running z1 10.244.0.138 6a8e24d8-8b3d-4039-5df0-692f0441e4b4 - true
log-api/96a84b7d-97da-4d2e-9a7a-aa8c2137d56a running z1 10.244.0.140 83039d61-f75d-4ce2-7aca-7d4eb76a9793 minimal true
nats/fd2949bd-9a5c-4bb0-8405-12d95aec70cb running z1 10.244.0.128 b0c76f83-99fe-41bf-7be0-23b80735f26d minimal true
router/d6b91efe-00c3-41db-94f9-250fdbbbde8f running z1 10.244.0.34 0d46b96a-5f9f-4bd2-7ce1-30ff9cf2f150 - true
scheduler/ae398dda-5bb2-4736-aa2e-a8ff3a0a88fb running z1 10.244.0.137 7e96af9d-a290-4630-442c-2314a06b2d91 - true
singleton-blobstore/9990d85e-5b2c-4de0-a9e5-d9004b4e4cfe running z1 10.244.0.133 e944cc59-593a-4e1b-663d-daecb8e228c8 small true
tcp-router/2bce23a1-7e88-43cb-af60-3a52c6d8849f running z1 10.244.0.136 2444bda0-2fff-448e-5514-5157555bef95 - true
uaa/8acb45cd-039c-40d1-8d42-9ea04c57bf68 running z1 10.244.0.132 d7977e63-3f90-41d0-7ab2-6771905dd670 minimal true

jobs on uaa
uaa/8acb45cd-039c-40d1-8d42-9ea04c57bf68:~# monit summary
The Monit daemon 5.2.5 uptime: 18h 29m

Process 'uaa' running
Process 'route_registrar' running
Process 'statsd_injector' running
Process 'loggregator_agent' running
Process 'loggr-expvar-forwarder' running
Process 'bosh-dns' running
Process 'bosh-dns-resolvconf' running
Process 'bosh-dns-healthcheck' running
System 'system_localhost' running

I queried some logs from uaa:
cat /var/vcap/sys/log/uaa/post-start.stderr.log
^M 0 0 0 0 0 0 0 0 --:--:-- --:--:-- --:--:-- 0curl: (7) Failed to connect to 127.0.0.1 port 8080: Connection refused
% Total % Received % Xferd Average Speed Time Time Time Current
Dload Upload Total Spent Left Speed
^M 0 0 0 0 0 0 0 0 --:--:-- --:--:-- --:--:-- 0curl: (7) Failed to connect to 127.0.0.1 port 8080: Connection refused
% Total % Received % Xferd Average Speed Time Time Time Current
Dload Upload Total Spent Left Speed
^M 0 0 0 0 0 0 0 0 --:--:-- --:--:-- --:--:-- 0curl: (7) Failed to connect to 127.0.0.1 port 8080: Connection refused

I'm not seeing any error logs in  /var/vap/sys/log/uaa/uaa.log

**Comment by ionphractal** [comment]
@forexamplelyr 
We got a similar issue. Would you please check `uaa.log`? In our case we saw db migration was failing with

```
[2020-02-17 14:23:48.211] uaa - 15 [main] .... ERROR --- DbMigrate: Migration of schema "public" to version 4.99.1575367461 - revocable token index [non-transactional] failed! Please restore backups and roll back database and code!
...
Migration V4_99_1575367461__revocable_token_index.sql failed
------------------------------------------------------------
SQL State  : 42601
Error Code : 0
Message    : ERROR: syntax error at or near "NOT"
  Position: 30
Location   : org/cloudfoundry/identity/uaa/db/postgresql/V4_99_1575367461__revocable_token_index.sql (/var/vcap/jobs/uaa/file:/var/vcap/data/uaa/tomcat/webapps/ROOT/WEB-INF/lib/cloudfoundry-identity-server-74.13.0.jar!/org/cloudfoundry/identity/uaa/db/postgresql/V4_99_1575367461__revocable_token_index.sql)
Line       : 1
Statement  : CREATE INDEX CONCURRENTLY IF NOT EXISTS revocable_tokens_user_id_client_id_response_type_identity__idx on revocable_tokens(user_id, client_id, response_type, identity_zone_id)
...
```
In our case, the database was postgresql 9.4 which is not supported since https://github.com/cloudfoundry/uaa/releases/tag/v74.13.0 anymore (-> "Added a database index on revocable tokens table submitted via PR #1154").

**Comment by strehle** [comment]
@ionphractal postgresql 9.4 is set EOL, so need to upgrade to a new version anyhow
see
https://www.postgresql.org/about/news/2011/
https://www.postgresql.org/support/versioning/

**Comment by ionphractal** [comment]
@strehle Yes, it is, I know. Some corporate environments are not so fast with updates and it's not that the PostgreSQL 9.4 will just explode because it is EOL. Also there was no notice in the release notes so far. So I only wanted to say "this could be it for you, too, please double check your logs". ;) Of course updating postgres is the best way to go forward.

**Comment by siennathesane** [comment]
@strehle I agree with @ionphractal, not having a warning of a minimum required version with an update isn't the best optics and the cloud.gov team was bitten by this today, actually. Granted, we're going to upgrade RDS and not have this problem, but still, a warning and requirement would be appreciated.

**Comment by shamus** [comment]
Oh hi. Thanks for reporting this issue. We totally feel your pain. I don't believe this change was totally intentional. However, it is now driving @dbeneke and myself to determine what database support can be expected of the UAA.

That said, we agree that intentional changes like this should be communicated via the release notes so you're not exposed to unfortunate surprises like this.

It sounds like you're planning to upgrade your postgres instance @mxplusb so you're not blocked by this change?  @ionphractal are you impacted at all currently?

**Comment by siennathesane** [comment]
Thinking about it, I would be okay with only the currently supported releases being supported with breaking changes being called out as they come up.

**Comment by ionphractal** [comment]
@shamus For our environments that don't use an RDS-like service we decided not to update UAA until we have a migration path (currently work in progress). We're not directly impacted at the moment but if there were CVEs in the meanwhile, we would be affected.

**Comment by shamus** [comment]
Closing this issue. I'm uncertain if the original poster's issue has been take care of but I know the later postgres issue has been resolved.

Please feel free to file another issue if we've missed something!
